code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -129,8 +129,20 @@ proc/flame_radius(radius = 1, turf/T) //~Art updated fire.
 		radius = 5
 	for(var/obj/flamer_fire/F in range(radius,T)) // No stacking flames!
 		cdel(F)
-	new /obj/flamer_fire(T, 5 + rand(0,11), 15, null, radius)
-
+	new /obj/flamer_fire(T, rand(20, 30), rand(20, 30), "red", radius)
+	for(var/mob/living/carbon/M in range(radius, T))
+		if(isXeno(M))
```

Comment:
This sequence of code is repeated. A proc would make sense.

Reply:
This is done.